xla/service/gpu/transforms/cudnn_fused_conv_rewriter.cc

```diff
@@ -488,83 +564,138 @@ void CaptureConvGraphRecursive(HloInstruction* instr,
+        ++num_existing_users;
       }
       continue;
     }
     // Scale
-    if (Match(user, m::MultiplyAnyOrder(&op, m::Op(&operand0),
+    if (Match(user, m::MultiplyAnyOrder(&op0, m::Op(&operand0),
                                         m::Broadcast(m::Op(&operand1)))) &&
         ShapeUtil::IsScalar(operand1->shape())) {
-      if (graph_string.AppendOp("scale", op, {operand0, operand1})) {
+      if (graph_string.AppendOp("scale", op0, {operand0, operand1})) {
         operands.push_back(operand1);
-        num_linear_users++;
+        ++num_new_users;
         CaptureConvGraphRecursive(user, operands, aux_outputs, graph_string,
-                                  visited_instrs, final_instr);
+                                  visited_instrs, final_instr, num_endpoints);
       }
       continue;
     }
     // Inverse Scale
-    if (Match(user, m::Divide(&op, m::Op(&operand0),
+    if (Match(user, m::Divide(&op0, m::Op(&operand0),
                               m::Broadcast(m::Op(&operand1)))) &&
         ShapeUtil::IsScalar(operand1->shape())) {
-      if (graph_string.AppendOp("invscale", op, {operand0, operand1})) {
+      if (graph_string.AppendOp("invscale", op0, {operand0, operand1})) {
         operands.push_back(operand1);
-        num_linear_users++;
+        ++num_new_users;
         CaptureConvGraphRecursive(user, operands, aux_outputs, graph_string,
-                                  visited_instrs, final_instr);
+                                  visited_instrs, final_instr, num_endpoints);
       }
       continue;
     }
     // ReLU
-    if (Match(user, m::MaximumAnyOrder(&op, m::Op(&operand0),
+    if (Match(user, m::MaximumAnyOrder(&op0, m::Op(&operand0),
                                        m::Broadcast(m::ConstantScalar(0))))) {
-      if (graph_string.AppendOp("relu", op, {operand0})) {
-        num_linear_users++;
+      if (graph_string.AppendOp("relu", op0, {operand0})) {
+        ++num_new_users;
         CaptureConvGraphRecursive(user, operands, aux_outputs, graph_string,
-                                  visited_instrs, final_instr);
+                                  visited_instrs, final_instr, num_endpoints);
       }
       continue;
     }
     //  Maximum of the absolute value (Amax) following ReLU (elided Abs) -- not
     //  a linear user
-    if (Match(user, m::Reduce(&op, m::Op(&operand0), m::Op())) &&
-        graph_string.OpInGraph(operand0->unique_id(), "relu") &&
-        AppliesMaxReduce(op)) {
-      if (graph_string.AppendOp("amax", op, {operand0})) {
-        aux_outputs.push_back(op);
-        num_nonlinear_users++;
+    if (Match(user, m::Reduce(&op0, m::Op(&operand0), m::Op())) &&
+        graph_string.OpInGraph(operand0, "relu") && AppliesMaxReduce(op0)) {
+      if (graph_string.AppendOp("amax", op0, {operand0})) {
+        aux_outputs.push_back(op0);
+        ++num_new_users;
+      }
+      continue;
+    }
+    // ReLU6, represented in the cuDNN graph as minimum(ReLU, 6)
```

Comment:
You should mention how its represented in the HLO graph as well. And same for ELU and Leaky Relu